product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.2-provision-user-using-SCIM1.1/src/test/java/org/wso2/identity/scenarios/test/scim/ScenarioTestBase.java

```diff
@@ -0,0 +1,98 @@
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.identity.scenarios.test.scim;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class ScenarioTestBase {
+
+    private static final String INPUTS_LOCATION = System.getenv("DATA_BUCKET_LOCATION");
+    private static final String INFRASTRUCTURE_PROPERTIES = "infrastructure.properties";
+    private static final String DEPLOYMENT_PROPERTIES = "deployment.properties";
+    private static final String JOB_PROPERTIES = "testplan-props.properties";
+    private static final Log LOG = LogFactory.getLog(ScenarioTestBase.class);
+
+    public static final String MGT_CONSOLE_URL = "MgtConsoleUrl";
+    public static final String CARBON_SERVER_URL = "CarbonServerUrl";
+    public static final String IS_HTTP_URL = "ISHttpUrl";
+    public static final String IS_HTTPS_URL = "ISHttpsUrl";
+    public static final String ADMIN_USERNAME = "admin";
+    public static final String ADMIN_PASSWORD = "admin";
+
+    protected static final int ARTIFACT_DEPLOYMENT_WAIT_TIME_MS = 120000;
+    protected static final String RESOURCE_LOCATION = System.getProperty("framework.resource.location");
+
+    /**
+     * This is a utility method to load the deployment details.
+     * The deployment details are available as key-value pairs in {@link #INFRASTRUCTURE_PROPERTIES},
+     * {@link #DEPLOYMENT_PROPERTIES}, and {@link #JOB_PROPERTIES} under the
+     * {@link #INPUTS_LOCATION}.
+     * <p>
+     * This method loads these files into one single properties, and return it.
+     *
+     * @return properties the deployment properties
+     */
+    public static Properties getDeploymentProperties() {
+
+        Path infraPropsFile = Paths.get(INPUTS_LOCATION + File.separator + INFRASTRUCTURE_PROPERTIES);
+        Path deployPropsFile = Paths.get(INPUTS_LOCATION + File.separator + DEPLOYMENT_PROPERTIES);
+        Path jobPropsFile = Paths.get(INPUTS_LOCATION + File.separator + JOB_PROPERTIES);
+
+        Properties props = new Properties();
+        loadProperties(infraPropsFile, props);
+        loadProperties(deployPropsFile, props);
+        loadProperties(jobPropsFile, props);
+        return props;
+    }
+
+    private static void loadProperties(Path propsFile, Properties props) {
+
+        if (!Files.exists(propsFile)) {
+            LOG.warn("Deployment property file not found: " + propsFile);
+            return;
+        }
+        try (InputStream propsIS = Files.newInputStream(propsFile)) {
```

Comment:
Don't we need to close this input stream?

Reply:
Since this is a try-with-resources statement the stream will be automatically closed.